src/rate-limit-view/rate-limit-view.ts

```diff
@@ -0,0 +1,89 @@
+import COLORS from "../data/rate-limit-view-colors.json";
+
+interface RateLimitColors {
+	negative: string,
+	positive: string[]
+}
+
+const RATE_LIMIT_COLORS: RateLimitColors = COLORS;
+const RATE_LIMIT = 333;
+const COLOR_INTERVAL = 20;
+const COLOR_MAX_MS = RATE_LIMIT_COLORS.positive.length * COLOR_INTERVAL;
+
+export class RateLimitView {
+	static oldtime = Date.now();
+	static newtime = Date.now();
+	static bgColor = "";
+	static fontColor = "";
+	static isDelta = false;
+	static isEnabled = false;
+
+	constructor(Options: any) {
+		$("<span id=ratelim class=ratelimit>RL: <span id=rate></span></span>").insertBefore(".CANT_REPLY");
+		$(".ratelimit").css({
+			background: "transparent",
+			display: "none",
+			float: "right",
+			"font-size": "smaller",
+			"margin-left": "10px",
+			"margin-top": "5px",
+		});
+		this.initOptions(Options);
+	}
+
+	initOptions(Options: any): void {
+		Options.addSelect({
+			default: 0,
+			help: "Rate limit viewer by post. Not real, just estimate lol",
+			label: "RATE LIMIT VIEW",
+			onchange: RateLimitView.onChange.call(this),
```

Comment:
I think `this` will be the `RateLimitReview` rather than the element as it was before. Does this work still?

Reply:
Oh yeah it is.
I thought it still worked, but the `this.val()` call errored while compiling.

Reply:
I'll move the logic back here for now. That works, although it requires an alias for `this` now that some variables moved to the class.